### ISSUE TITLE
Can't dynamically update the value of a BIP element

### ISSUE THREAD

**Issue by noctivityinc** [bernat/best_in_place#439]
I have two tabs on a page.  Inside each tab I have the same document title.  I want to let the user update, using BIP, the title from either page.

They can do this but they are not in sync.  I hooked into the ajax:success callback and tried something like:

    $('#firstPage .best_in_place').text(data.label).attr('data-bip-value', data.label).attr('data-bip-original-content', data.label)

which updated the attributes but when I click on the element from the first page the original values are still used.

I've searched through all the docs but can't seem to find a simple way to do something like:

     $('#firstPage .best_in_place').best_in_place({value: 'new value'}) 

or the like to update the value of the field.  How can you dynamically update the value of a BIP element?

**Comment by davidrouten** [comment]
Noctivity, I'm having the same issue with a similar concept. There are at times two bips on screen at once for the same attribute for the same instance. Everything works extremely smoothly, but if you update one there doesn't seem to be any way (I tried doing the exact same thing you did, hooking into the ajax:success call) to update the unchanged bip so that it will reflect the data saved to the database, and not what was previously entered before the change.

After digging around, it appears that the actual value used is stored in a JS object and that that is where it is pulled from, but I'm not strong enough in JS to find a solution without spending an entire day trying to figure out how to access and change that info.

Would appreciate any help!

**Comment by vzelenko** [comment]
Use the editor object:

var editor = $('#firstPage .best_in_place').data('bestInPlaceEditor');

Then in JS debugger you can see what values it contains.

I have also replaced 'data-bip-*' attribute values where appropriate, but not sure if that at all matters.

**Comment by stevenspiel** [comment]
thank you @vzelenko! That was just what I needed!

**Comment by bevali** [comment]
Is there documentation on how to use the editor object to change the attributes? I want to dynamically change the dropdown option for a form field based on the value of another form field. I was able to create the editor and look at the object. I tried editor.collection = "[['a','a'],['b','b']]". That saved, but the form doesn't change. I've also tried setting data-bip-collection.

**Comment by vzelenko** [comment]
Yes, I could not get it either.  Selects are a bit tricky, as you have to rebuild the element.  I ended up using dual-select control of my own...  

Now, consider for your case an autocomplete solution  (like typeahead) that might make your life easier.

**Comment by bevali** [comment]
Thanks @vzelenko. I appreciate your help. I might try to make my own too, but was hoping there was an easier way.
